src/TgSharp.Core/Network/MtProtoSender.cs

```diff
@@ -388,82 +406,58 @@ private bool HandleMsgDetailedInfo(ulong messageId, int sequence, BinaryReader m
+                    case 19:
+                        throw new InvalidOperationException("container msg_id is the same as msg_id of a previously received message (this must never happen)");
+                    case 20:
+                        throw new InvalidOperationException("message too old, and it cannot be verified whether the server has received a message with this msg_id or not");
+                    case 32:
+                        throw new InvalidOperationException("msg_seqno too low (the server has already received a message with a lower msg_id but with either a higher or an equal and odd seqno)");
+                    case 33:
+                        throw new InvalidOperationException(" msg_seqno too high (similarly, there is a message with a higher msg_id but with either a lower or an equal and odd seqno)");
+                    case 34:
+                        throw new InvalidOperationException("an even msg_seqno expected (irrelevant message), but odd received");
+                    case 35:
+                        throw new InvalidOperationException("odd msg_seqno expected (relevant message), but even received");
+                    case 48:
+                        throw new InvalidOperationException("incorrect server salt (in this case, the bad_server_salt response is received with the correct salt, and the message is to be re-sent with it)");
+                    case 64:
+                        throw new InvalidOperationException("invalid container");
+
+                        throw new InvalidOperationException($"Unknown error code ({errorCode})");
+                }
+            }
+            catch (InvalidOperationException ex)
+            {
+                SetRequestException(requestId, ex);
             }
-            throw new NotImplementedException("This should never happens");
-            /*
-			logger.debug("bad_msg_notification: msgid {0}, seq {1}, errorcode {2}", requestId, requestSequence,
-						 errorCode);
-			*/
-            /*
-			if (!runningRequests.ContainsKey(requestId))
-			{
-				logger.debug("bad msg notification on unknown request");
-				return true;
-			}
-			*/
-
-            //OnBrokenSessionEvent();
-            //MTProtoRequest request = runningRequests[requestId];
-            //request.OnException(new MTProtoBadMessageException(errorCode));
 
             return true;
         }
 
-        private bool HandleBadServerSalt(ulong messageId, int sequence, BinaryReader messageReader, TLMethod request, CancellationToken token = default(CancellationToken))
+        private bool HandleBadServerSalt(ulong messageId, int sequence, BinaryReader messageReader)
         {
-            token.ThrowIfCancellationRequested();
-
-            uint code = messageReader.ReadUInt32();
-            ulong badMsgId = messageReader.ReadUInt64();
+            _ = messageReader.ReadUInt32();
```

Comment:
in HandlePong, you have put a comment explaining why this is ignored (_), can you put a comment here too?

Reply:
all those comments are mostly old I just ignored some of the unused variables

Reply:
maybe it's easy to figure out from looking at the spec again for a minute?